cmd/main_test.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"encoding/json"
+	"os"
+	"regexp"
+	"strconv"
+	"testing"
+
+	"github.com/SUMUKHA-PK/Raft-Distributed-Consensus/types"
+)
+
+func TestMain(t *testing.T) {
+	//check if the config file exists
+	info, err := os.Stat("../server.config.json")
+	if os.IsNotExist(err) {
+		t.Fatalf("Config file does not exist")
+	}
+	//check if there is no read permission to config file
+	mask := info.Mode()
+	if mask&(1<<2) == 0 {
```

Comment:
Is this necessary? What happens when the file is read protected and its opened? If theres no explicit error for this, include this test.

Reply:
You are checking if the read bit is on before opening. What else needs to be done?

Reply:
I'm telling we might not need an explicit test if the file opener already throws an error.

Check whether that happens, if it doesn't let it stay.